Samples/BackgroundBlur/BackgroundBlur/Helpers/CSampleQueue.cpp

```diff
@@ -0,0 +1,296 @@
+#include "CSampleQueue.h"
+// TODO: Maybe move the GUID to root's common.h
+#include "../TransformAsync.h"
+#include "common.h"
+
+
+class CSampleQueue::CNode
+{
+public:
+    IMFSample*  pSample;
+    CNode*      pNext;
+
+    CNode(void)
+    {
+        pSample = NULL;
+        pNext   = NULL;
+    }
+    ~CNode(void)
+    {
+        SAFE_RELEASE(pSample);
+        pNext   = NULL;
+    }
+};
+
+HRESULT CSampleQueue::Create(
+    CSampleQueue**   ppQueue)
+{
+    HRESULT         hr          = S_OK;
+    CSampleQueue*   pNewQueue   = NULL;
+
+    do
+    {
+        if(ppQueue == NULL)
+        {
+            hr = E_POINTER;
+            break;
+        }
+
+        pNewQueue = new CSampleQueue();
+        if(pNewQueue == NULL)
+        {
+            hr = E_OUTOFMEMORY;
+            break;
+        }
+
+        (*ppQueue) = pNewQueue;
+        (*ppQueue)->AddRef();
+    }while(false);
+
+    SAFE_RELEASE(pNewQueue);
+
+    return hr;
+}
+
+ULONG CSampleQueue::AddRef(void)
+{
+    return InterlockedIncrement(&m_ulRef);
+}
+
+HRESULT CSampleQueue::QueryInterface(
+    REFIID riid,
+    void** ppvObject)
+{
+    HRESULT hr = S_OK;
+
+    do
+    {
+        if(ppvObject == NULL)
+        {
+            hr = E_POINTER;
+            break;
+        }
+
+        if(riid == IID_IUnknown)
+        {
+            *ppvObject = (IUnknown*)this;
+        }
+        else
+        {
+            *ppvObject = NULL;
+            hr = E_NOINTERFACE;
+            break;
+        }
+
+        AddRef();
+    }while(false);
+
+    return hr;
+}
+
+ULONG CSampleQueue::Release(void)
+{
+    ULONG   ulRef = 0;
```

Comment:
Nit: The spacing is inconsistent in different places, I suggest use single space to separate symbols across the file.